src/analysis/destruct.ml

```diff
@@ -588,9 +588,42 @@ module Conv = struct
     (ps, constrs, labels)
 end
 
+let need_recover_labeled_args = function
+  | Parsetree.Pexp_construct ({loc; txt = Longident.Lident ctor}, Some e) ->
+    (* If the internal construction is ghosted, then the expression must be
+       re-labelled. *)
+    if String.equal "Some" ctor && loc.loc_ghost then Some e else None
+  | _ -> None
```

Comment:
I don't understand why this is needed :-) 
What would happen in the testsuite examples without it ?

Reply:
```
$ $MERLIN single case-analysis -start 2:10 -end 2:15 \
  > -filename main.ml <<EOF
  > let foo ?(bar = 10) x = x
  > let () = foo ~bar:15 ()
  > EOF
  {
    "class": "return",
    "value": [
      {
        "start": {
          "line": 2,
          "col": 9
        },
        "end": {
          "line": 2,
          "col": 23
        }
      },
      "match foo ?bar:(Some 15) () with | () -> _"
    ],
    "notifications": []
  }
```

and 

```
$ $MERLIN single case-analysis -start 2:10 -end 2:15 \

  $ $MERLIN single case-analysis -start 2:10 -end 2:15 \
  > -filename main.ml <<EOF
  > let foo ?bar x = x
  > let () = foo ~bar:10 ()
  > EOF
  {
    "class": "return",
    "value": [
      {
        "start": {
          "line": 2,
          "col": 9
        },
        "end": {
          "line": 2,
          "col": 23
        }
      },
      "match foo ?bar:(Some 10) () with | () -> _"
    ],
    "notifications": []
  }

```